### PR TITLE
Fix #1258

### LINKED ISSUE #1258: Struct closing brace indentation preserved
I've noticed that the indentation on the closing brace on structs seems to be preserved. 

I forked rustfmt and committed a failing test which illustrates this:

https://github.com/casey/rustfmt/commit/fb79ecf6de98ab3ba1f0bd9a666f7239d07ec410

This is the error message:

```
Mismatch at tests/source/structs.rs:161:
 }⏎
 ⏎
 struct Foo {}⏎
-struct Foo {}⏎
+struct Foo {⏎
+  }⏎
 ⏎
 struct Foo();⏎
```

### PR DESCRIPTION
**Pull request by sinkuu**
#1258

Checking if newline exists to preserve this code as is (tested in `issue-977.rs`).

```rust
struct Foo { /* comment */ }
```

### CODE REVIEW

File: src/items.rs [hunk 1]
```diff
@@ -839,7 +839,18 @@ fn format_struct_struct(context: &RewriteContext,
 
     // FIXME(#919): properly format empty structs and their comments.
     if fields.is_empty() {
-        result.push_str(&context.snippet(mk_sp(body_lo, span.hi)));
+        let snippet = context.snippet(mk_sp(body_lo, span.hi - BytePos(1)));
+        if snippet.trim().is_empty() {
+            // `struct S {}`
+        } else if snippet.trim_right_matches(&[' ', '\t'][..]).ends_with('\n') {
```
Comment (nrc) [inline src/items.rs]: I don't think the `[..]` is necessary here, or might be better expressed with `*`.

File: tests/source/structs.rs [hunk 2]
```diff
@@ -153,4 +153,9 @@ struct Issue677 {
 }
 
 struct Foo {}
+struct Foo {
+    }
+struct Foo {
+    // comment
+    }
```
Comment (nrc) [inline tests/source/structs.rs]: Could you add a test which has multiple blank lines and trailing whitespace after the comment please? Also a test which should end up on one line (I assume this means a comment with trailing whitespace, but no newline).

File: src/items.rs (hunk 1 above)
Comment (sinkuu) [inline src/items.rs]: `trim_right_matches` takes a `Pattern` and thus deref coersion doesn't work here IIUC.

File: tests/source/structs.rs (hunk 2 above)
Comment (sinkuu) [inline tests/source/structs.rs]: Added tests.

File: src/items.rs (hunk 1 above)
Comment (nrc) [inline src/items.rs]: True, you could use ` as &[_]` which is a bit more explicit about what is going on, although a little more verbose. I guess `[..]` is ok.

**Comment by nrc** [comment]
Thanks!
